diff --git a/dist/metro/index.js b/dist/metro/index.js
index c91ba6adf049dd5c6eec1ec8301ecc0802f41bfc..6dbd1edd230593a1d4bb8b552bf7d69ddb5a4ee9 100644
--- a/dist/metro/index.js
+++ b/dist/metro/index.js
@@ -176,18 +176,25 @@ async function startCSSProcessor(filePath, platform, isDev, { input, getCSSForPl
                 ? css
                 : getNativeJS((0, css_to_rn_1.cssToReactNativeRuntime)(css, options), debug)));
             debug(`virtualStyles.emit ${platform}`);
+            const metadata = {
+                modifiedTime: Date.now(),
+                size: 1,
+                type: "virtual",
+            };
             haste.emit("change", {
+                changes: {
+                    addedFiles: new Map(),
+                    modifiedFiles: new Map([[filePath, metadata]]),
+                    removedFiles: new Map(),
+                },
                 eventsQueue: [
                     {
                         filePath,
-                        metadata: {
-                            modifiedTime: Date.now(),
-                            size: 1,
-                            type: "virtual",
-                        },
+                        metadata,
                         type: "change",
                     },
                 ],
+                rootDir: "",
             });
         }).then((css) => {
             debug(`virtualStyles.initial ${platform}`);
diff --git a/dist/runtime/native/render-component.js b/dist/runtime/native/render-component.js
index 9f4966b..9c1f1b3 100644
--- a/dist/runtime/native/render-component.js
+++ b/dist/runtime/native/render-component.js
@@ -133,8 +133,13 @@ function stringify(object) {
         seen.add(value);
         const newValue = Array.isArray(value) ? [] : {};
-        for (const entry of Object.entries(value)) {
-            newValue[entry[0]] = replace(entry[0], entry[1]);
+        for (const key of Object.keys(value)) {
+            try {
+                newValue[key] = replace(key, value[key]);
+            }
+            catch (error) {
+                newValue[key] = `[Throws: ${error?.message ?? String(error)}]`;
+            }
         }
         seen.delete(value);
         return newValue;
     }, 2);
 }
diff --git a/src/runtime/native/render-component.tsx b/src/runtime/native/render-component.tsx
index 1c71c6e..94d0be2 100644
--- a/src/runtime/native/render-component.tsx
+++ b/src/runtime/native/render-component.tsx
@@ -253,8 +253,12 @@ function stringify(object: any) {
 
       const newValue: any = Array.isArray(value) ? [] : {};
 
-      for (const entry of Object.entries(value)) {
-        newValue[entry[0]] = replace(entry[0], entry[1]);
+      for (const key of Object.keys(value)) {
+        try {
+          newValue[key] = replace(key, value[key]);
+        } catch (error: any) {
+          newValue[key] = `[Throws: ${error?.message ?? String(error)}]`;
+        }
       }
 
       seen.delete(value);
 
       return newValue;
